app/main.py

```diff
@@ -1,13 +1,31 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {"line": error["line_number"], "column": error["column_number"],
+            "message": error["text"], "name": error["code"],
+            "source": "flake8"}
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {"errors": [{"line": i["line_number"], "column": i["column_number"],
+                        "message": i["text"], "name": i["code"],
+                        "source": "flake8"}for i in errors],
+            "path": file_path, "status": "failed"}
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [
+        {
+            "errors": [
+                {
+                    "line": error["line_number"],
+                    "column": error["column_number"],
+                    "message": error["text"],
+                    "name": error["code"],
+                    "source": "flake8",
+                }
+                for error in errors
```

Comment:
The variable `errors` is used here, but it is not defined within the loop context. You should ensure that `errors` is correctly defined or passed in the loop, possibly by iterating over `linter_report.items()` as `(path, errors)`.